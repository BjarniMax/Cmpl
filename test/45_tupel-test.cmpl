%arg -noOutput


parameters:

	routes  := set( [1,1], [1,2], [1,4] , [2,2], [2,3] ,[2,4] , [3,1] , [3,3] );
	
	plants  := 1(1)3;
	centers := 1(1)4;

	s[plants]  := (5000,6000,2500);
	d[centers] := (6000,4000,2000,1500);

	c[routes] := ( 3, 2, 6, 5, 2, 3, 2, 4 );
	
	
	ts := set( [1,1], [1,3], [2,3], [3,3], [3,2] );
	t2 := set( [3,3], [2,2], [1,1] );
	
	A[ts] := (4, 67, 44, -5, 0);
	

 	echo plants + centers;
	echo  plants * centers;

	echo len(routes); 
	echo len(plants); 
	
	{k in  defset(s[]): echo " k " + k; }
	
	echo " tt " + [1,1]; 
	{ t in  routes: echo " tk " +  t; } 
	
		
	echo "max s " +max(s[plants]);
	
	echo "max c " + max(c[routes]);
	echo max(c);
	
	echo "min c " + min(c[,]);
	echo min(c);
	
	echo count(c[routes]);
	echo count(c[[1,1]]);

	echo max(A[t2]);
	echo min(A[t2]);
	
	echo max(A[ts]);
	echo min(A[ts]);
	
	echo count(A[t2]);
	 
 