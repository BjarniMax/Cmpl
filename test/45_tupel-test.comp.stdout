1..4
1..3
8
3
 k 1
 k 2
 k 3
 tt [1, 1]
 tk [1, 1]
 tk [1, 2]
 tk [1, 4]
 tk [2, 2]
 tk [2, 3]
 tk [2, 4]
 tk [3, 1]
 tk [3, 3]
max s 6000
max c 6
6
min c 2
2
8
1
4
-5
67
-5
2
